RFCXXX-Native-Json-Adapters.md

```diff
@@ -0,0 +1,258 @@
+This means that rather than just capturing the parameter name and default value, we will need to be able to run arbitrary code (ala scriptblock)
+which includes the command line AST so the proper parameter values may be created.
+
+However, with an approach which provides for this complexity, we can continue to simplify the experience for our users to from:
+
+```powershell
+PS> docker --debug run --rm -it mcr.microsoft.com/powershell
+```
+
+to
+
+```powershell
+PS> docker run mcr.microsoft.com/powershell
+```
+
+or
+
+```powershell
+PS> Docker image ls --format='{{json .}}' | ConvertFrom-Json | Where-Object {$_.Repository -match "mcr"}| Format-Table Id,Repository,Tag,Size,CreatedAt
+
+ID           Repository                             Tag                     Size   CreatedAt
+--           ----------                             ---                     ----   ---------
+4e9617ada198 mcr.microsoft.com/powershell           preview-7.3-mariner-2.0 235MB  2022-09-01 11:48:17 -0700 PDT
+. . .
+```
+
+to this:
+
+```powershell
+PS> docker image ls | ? Repository -match "mcr"
+
+ID           Repository                             Tag                     Size   CreatedAt
+--           ----------                             ---                     ----   ---------
+4e9617ada198 mcr.microsoft.com/powershell           preview-7.3-mariner-2.0 235MB  2022-09-01 11:48:17 -0700 PDT
+. . .
+```
+
+### Follow-on Work
+
+A number of follow-on activities should be considered to bring continued improvement of the native command experience.
+
+#### Improved TabCompletion
+
+Current tab-completion (on Unix) is done via a somewhat fragile call to the native shell.
+Tab-completion on Windows must be implemented specifically.
+we should take the opportunity to improve our tooling around tab-completion to make it easier to create,
+and provide generalized help parsers (where possible) to generate tab-completion.
+`bash` provides a number of more generalized tab-completion, we should attempt to do the same for PowerShell users.
+
+For example, bash has a single completer all known hosts which is used by nearly 20 executables.
+We already will tab-complete for file system items,
+a small effort should be able to increase our coverage for other native executables.
+
+#### Improved Formatting Tooling
+
+Once text output is converted to objects,
+we should improve the process for creating formatting directives.
+When creating a formatting file,
+the user needs to manually author the `XML` and then call `update-formatdata` to add that configuration to the current session.
+Usually, formatting files are associated with a module and distributed in this way.
+With stand-alone native applications, it means that providing for reasonable formatting will require an additional file.
+A more dynamic approach for formatting would improve the experience for the user.
+
+## Open Questions
```

Comment:
How do we handle errors? For example, in the case the json is not formatted correctly/